lib/matplotlib/backends/backend_agg.py

```diff
@@ -562,24 +580,22 @@ def print_jpg(self, filename_or_obj, *args, **kwargs):
             *progressive*: If present, indicates that this image
                 should be stored as a progressive JPEG file.
             """
-            buf, size = self.print_to_buffer()
+            buf, (w, h) = self.print_to_buffer()
+            buf = np.array(memoryview(buf)).reshape((w, h, 4))
+
             if kwargs.pop("dryrun", False):
                 return
-            # The image is "pasted" onto a white background image to safely
-            # handle any transparency
-            image = Image.frombuffer('RGBA', size, buf, 'raw', 'RGBA', 0, 1)
-            color = mcolors.colorConverter.to_rgb(
-                rcParams.get('savefig.facecolor', 'white'))
-            color = tuple([int(x * 255.0) for x in color])
-            background = Image.new('RGB', size, color)
-            background.paste(image, image)
+
+            # Flatten RGBA image to safely handle transparent regions
+            buf = mimage.flatten_rgba(buf)
+            img = Image.frombuffer('RGBA', (w, h), buf, 'raw', 'RGBA', 0, 1)
```

Comment:
The images are in row-major order (i.e. as English text is read), and Numpy dimensions are in the order of "slowest moving" to "fastest moving" dimension, thus (h, w, 4).  It's a Numpy/array programming convention.  PIL, as an image library, not an array library, uses the more common image convention (w, h).